feature: add back calendar entities

LeonelCeballos18 opened the pull request:
## What kind of PR is this?
- [x] Feature

## Description
adds appointment entity
adds appointment service entity
adds customer entity
adds service entity
adds relations between entities
adds migrations
adds all entities in data source file

## Ticket id
SCRUM-37-Back | Calendar
https://barbershop-app.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-37

## Useful resources
<img width="775" alt="Screenshot 2025-02-10 at 12 49 48" src="https://github.com/user-attachments/assets/2ac175db-0dba-4148-8343-aa05db484112" />

LeonelCeballos18 committed:
feature: add back calendar entities

adds appointment entity
adds appointment service entity
adds customer entity
adds service entity
adds relations between entities
adds migrations
adds all entities in data source

cmagallon1 commented on src/migration/1738950139248-add_service_entity.ts:
```diff
@@ -0,0 +1,15 @@
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "service" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, "price" integer NOT NULL, "duration" integer NOT NULL, "description" character varying NOT NULL, CONSTRAINT "PK_85a21558c006647cd76fdce044b" PRIMARY KEY ("id"))`,
```
for prices or any values in the database that involve money it's always better to use `decimal` rather than float/integer; It helps to be more accurate when doing calculations

cmoreno357 replied:
in TypeORM with PostgreSQL there's also the type `money`, decimal would work but I'd rather use the former due to semantics.

cmagallon1 replied:
yeah sure, didn't know nestjs had a specific value for this specific case, but totally agree that if we have something for that we should use it

cmagallon1 replied:
I think here you can find all the available entities @LeonelCeballos18 @cesarj45 : https://github.com/typeorm/typeorm/blob/b9263d50d6281ed93cd74846527cbbb099b1891f/docs/entities.md

CC. @cmoreno357

cmoreno357 replied:
> yeah sure, didn't know nestjs had a specific value for this specific case, but totally agree that if we have something for that we should use it

haha yea, it's actually not just NestJS, it's a native PostgreSQL data type

cmagallon1 commented on src/migration/1738958324399-add_Appointment_AppointmentService_and_relations.ts:
```diff
@@ -0,0 +1,20 @@
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "appointment_service" ("id" SERIAL NOT NULL, "appointmentId" integer, CONSTRAINT "PK_a170b01d5845a629233fb80a51a" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "appointment" ("id" SERIAL NOT NULL, "status" character varying NOT NULL, "scheduled_at" TIMESTAMP NOT NULL, "user_id" integer NOT NULL, "total_price" integer NOT NULL, "duration" integer NOT NULL, "customerId" integer NOT NULL, CONSTRAINT "PK_e8be1a53027415e709ce8a2db74" PRIMARY KEY ("id"))`);
```
same here with `total_price` better use `decimal` for these kind of fields

LeonelCeballos18 replied:
I was thinking set `total_price` as `decimal` but the cost of the services it'll be integer, it's that's why I decided to set it as integer, but even knowing that the prices going to be integer is it better set it as `decimal`?

cmagallon1 replied:
yes, it's always better to set it as `decimal` otherwise it can lead to precision issues. If you need to return the value as `integer` you can remove the decimals or round the value later in the code

cmoreno357 replied:
Same thing, checkout the type `money`

cmoreno357 commented:
💭 Just a heads up... In regards to [SCRUM-37](https://barbershop-app.atlassian.net/browse/SCRUM-37?atlOrigin=eyJpIjoiOGM0MmQ5MWNmMzZiNGYzYTlmMjIyNDM4NjE5YmMwMGUiLCJwIjoiaiJ9), Specifically on `scenario 5`, shouldn't this PR include some insert `POST` endpoints in controllers?

cmagallon1 commented on src/data-source.ts:
```diff
@@ -1,5 +1,9 @@
 import 'reflect-metadata';
 import { DataSource } from 'typeorm';
+import { Appointment } from './entity/Appointment';
```
💭 as a suggestion, you can have a file called `index.ts` file in the root of the entity folder where you can add all your classes:

index.ts:

```js
  export { default as Appointment } from '....path';
  export { default as AppointmentService } from '...path'
  export { default as Customer } from '...path'
   ...
```

this way you can easily import the classes this way:

```js
import {
   Appointment,
   Customer,
   AppointmentService,
   ...
} from './entity'
```

cmoreno357 replied:
💅  yup, the index file is a common practice in Typescript, here's some example if you'd like to implement it.
https://medium.com/@bobjunior542/maximize-typescript-productivity-use-the-index-file-now-cac2f4f4d1ed#:~:text=How%20Does%20the%20Index%20File,defined%20in%20the%20same%20file.

mtopete1996 reviewed (approved):
Good enough for me :)

LeonelCeballos18 committed:
chore: Solving comments

change `total_price`field from appointments from int to float and `price` field from services from int to float
change relation customers and appointments one to one
adds index file which containts all the clases
